pkg/network/tracer/offsetguess.go

```diff
@@ -796,19 +803,33 @@ func newEventGenerator(ipv6 bool) (*eventGenerator, error) {
 	return eg, nil
 }
 
-func getUDP6Conn(ipv6 bool) (*net.UDPConn, error) {
+func getUDP6Conn(ipv6 bool) (c *net.UDPConn, err error) {
 	if !ipv6 {
 		return nil, nil
 	}
 
-	linkLocal, err := getIPv6LinkLocalAddress()
-	if err != nil {
-		// TODO: Find a offset guessing method that doesn't need an available IPv6 interface
-		log.Debugf("unable to find ipv6 device for udp6 flow offset guessing. unconnected udp6 flows won't be traced: %s", err)
-		return nil, nil
-	}
+	// it is necessary to run the following code in
+	// the root network namespace, as the network
+	// namespace where we're running may not have
+	// ipv6 enabled. we are here because the host
+	// does have ipv6 enabled (ipv6 bool is true).
+	// if system-probe's network namespace does
+	// not have ipv6 enabled, the below code will fail
+	// since none of the interfaces will have an
+	// ipv6 link local address assigned
```

Comment:
Is this part of the comment saying "if system-probe is running in a container, and that's container's network namespace doesn't have ipv6 enabled, then this won't work"?

Reply:
Essentially, yes.